overlord/confdbstate/confdbstate.go

```diff
@@ -521,3 +521,58 @@ func IsConfdbHook(ctx *hookstate.Context) bool {
 			strings.HasPrefix(ctx.HookName(), "save-view-") ||
 			strings.HasSuffix(ctx.HookName(), "-view-changed"))
 }
+
+func createLoadConfdbTasks(st *state.State, tx *Transaction, view *confdb.View) (*state.TaskSet, error) {
+	custodians, custodianPlugs, err := getCustodianPlugsForView(st, view)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(custodians) == 0 {
+		return nil, fmt.Errorf("cannot load confdb %s/%s: no custodian snap connected", view.Confdb().Account, view.Confdb().Name)
+	}
+
+	ts := state.NewTaskSet()
+	linkTask := func(t *state.Task) {
+		tasks := ts.Tasks()
+		if len(tasks) > 0 {
+			t.WaitFor(tasks[len(tasks)-1])
+		}
+		ts.AddTask(t)
+	}
+
+	// check for load-view and query-view hooks on custodians
+	for _, hookPrefix := range []string{"load-view-", "query-view-"} {
+		for _, name := range custodians {
+			plug := custodianPlugs[name]
+			custodian := plug.Snap
+			if _, ok := custodian.Hooks[hookPrefix+plug.Name]; !ok {
+				continue
+			}
+
+			const ignoreError = false
+			task := setupConfdbHook(st, name, hookPrefix+plug.Name, ignoreError)
+			linkTask(task)
+		}
+	}
+
+	if len(ts.Tasks()) == 0 {
+		// no hooks to run and not running from API (don't need task to populate)
+		// data in change so we can just read the databag synchronously
+		return nil, nil
+	}
+
+	// clear the ongoing tx from the state and unblock other writers waiting for it
+	clearTxTask := st.NewTask("clear-confdb-tx", "Clears the ongoing confdb transaction from state")
+	clearTxTask.Set("confdb-transaction", tx)
```

Comment:
I am a bit confused, Is the caller of `createLoadConfdbTasks` expected to clear the tx when no hooks are found for snap (i.e. returned task set is nil) but not when hooks are found?

Reply:
that's a good question, this looks odd out of context. The full context can be seen in https://github.com/canonical/snapd/compare/master...miguelpires:confdb-read?expand=1 In short, yes. When there are no hooks to run, a `snapctl get` can just read from the databag synchronously. So there's no long lived transaction that we need to eventually remove from state.

Reply:
Thanks for the explanation, I think it would be good to have a doc comment on the helper explaining this behavior.

Reply:
Added, thanks!